app/program/page.tsx

```diff
@@ -9,7 +9,7 @@ export default function DaoPage() {
   const params = useSearchParams();
   const programKey = params.get('programKey');
   return (
-    <Layout>
+    <Layout programKey={programKey}>
```

Comment:
I don't think it's super clean to do that way. I would create a TokenPrice component and useSearchParams inside this one, so nothing is passed to the layout comp

Reply:
The issue is where the component exists. In layout.. Ahh actually I see what you're saying there, the only thing is we don't need search params for ALL layouts... And then I'd have to pass that through... Dunno I feel like it's sloppy to get it done, vs break this dead cat open more...

Reply:
if you do params.get('programKey') and it's not in the path, then it would return undefined and you would not show the price

Reply:
Ahhh gotcha. I do have a check for that, where it just defaults to META so it does work. But I take your point.

Reply:
Fixed, ready for review.